packages/story-editor/src/app/story/effects/useLoadStory.js

```diff
@@ -111,6 +109,22 @@ function useLoadStory({ storyId, shouldLoad, restore, isDemo }) {
           };
         }
 
+        let publisherLogo = {
+          id: 0,
+          height: 0,
+          width: 0,
+          url: '',
+        };
+
+        if ('wp:publisherlogo' in embedded) {
+          publisherLogo = {
+            id: embedded['wp:publisherlogo'][0].id,
+            height: embedded['wp:publisherlogo'][0].media_details?.height,
+            width: embedded['wp:publisherlogo'][0].media_details?.width,
+            url: embedded['wp:publisherlogo'][0].source_url,
+          };
+        }
+
```

Comment:
There is some repeated code here. How about a function like this?

```
function getEmbeddedAttachment( field, embedded ) {
        if (field in embedded) {
          return {
            id: embedded[field][0].id,
            height: embedded[field][0].media_details?.height,
            width: embedded[field][0].media_details?.width,
            url: embedded[field][0].source_url,
          };
        }
        return {
          id: 0,
          height: 0,
          width: 0,
          url: '',
        };
}
```

Could also be used like in this https://github.com/google/web-stories-wp/pull/8815

Reply:
With just 2 such fields I don't think it's worth extracting this into a function at this point. I'd consider it when there should be more fields, when the benefit would be bigger.

Reply:
It could be used [here](https://github.com/google/web-stories-wp/pull/8815/files#diff-6eac1ab8374e1da9ea0278131b7244d9d7e8e879c544d282a38c278453cdbe75R86-R99) and [here](https://github.com/google/web-stories-wp/pull/8815/files#diff-6eac1ab8374e1da9ea0278131b7244d9d7e8e879c544d282a38c278453cdbe75R137-R150). 

So the function is cleaner,as there is no need for a let.

Reply:
Also could be used [here](https://github.com/google/web-stories-wp/blob/610bbea66f7e0d4361956cab79f279f70d0e3c1d/packages/dashboard/src/app/serializers/stories.js#L45)  and maybe even [here](https://github.com/google/web-stories-wp/blob/610bbea66f7e0d4361956cab79f279f70d0e3c1d/packages/stories-block/src/block/components/storiesPreview.js#L86)

Reply:
I think just using optional chaining would do the trick just as well without hiding anything of this logic behind a function living in some other package.

```js
const publisherLogo = {
	  id: embedded?.['wp:publisherlogo']?.[0].id || 0,
	  height: embedded?.['wp:publisherlogo']?.[0]?.media_details?.height || 0,
	  width: embedded?.['wp:publisherlogo']?.[0]?.media_details?.width || 0,
	  url: embedded?.['wp:publisherlogo']?.[0]?.source_url || '',
};
```